src/mol-io/common/string-like.ts

```diff
@@ -0,0 +1,572 @@
+    // toLocaleLowerCase(locales?: Intl.LocalesArgument): string;
+
+    // /** Returns a string where all alphabetic characters have been converted to uppercase, taking into account the host environment's current locale. */
+    // toLocaleUpperCase(locales?: Intl.LocalesArgument): string;
+
+    // /**
+    //  * Determines whether two strings are equivalent in the current or specified locale.
+    //  * @param that String to compare to target string
+    //  * @param locales A locale string or array of locale strings that contain one or more language or locale tags. If you include more than one locale string, list them in descending order of priority so that the first entry is the preferred locale. If you omit this parameter, the default locale of the JavaScript runtime is used. This parameter must conform to BCP 47 standards; see the Intl.Collator object for details.
+    //  * @param options An object that contains one or more properties that specify comparison options. see the Intl.Collator object for details.
+    //  */
+    // localeCompare(that: string, locales?: Intl.LocalesArgument, options?: Intl.CollatorOptions): number;
+}
+
+
+export type StringLike = string | String | CustomString
+
+export function isStringLike(s: unknown): s is StringLike {
+    return typeof s === 'string' || s instanceof String || (s as CustomString).__string_like__;
+}
+
+/** Try to convert `StringLike` to a primitive `string`. Might fail if the contents is longer that max allowed string length. */
+export function stringLikeToString(s: StringLike): string {
+    try {
+        return s.toString();
+    } catch (err) {
+        throw new Error(`Failed to convert StringLike object into string. This might be because the length ${s.length} exceeds maximum allowed string length ${MAX_STRING_LENGTH}. (${err})`);
+    }
+}
+
+/** Maximum allowed string length (might be bigger for some engines, but in Chrome and Node it is this). */
+const MAX_STRING_LENGTH = 536_870_888;
+
+/** String chunk size to be used by `ChunkedBigString`. */
+const STRING_CHUNK_SIZE = 2 ** 28; // largest power of 2 smaller than MAX_STRING_LENGTH
+// const STRING_CHUNK_SIZE = 8; // DEBUG, TODO use a sensible value, close to MAX_STRING_LENGTH (e.g. 2**28)
+
+
+export class ChunkedBigString implements CustomString {
+    readonly __string_like__: true;
+    private _chunks: string[] = [];
+
+    readonly [index: number]: string; // implemented in the constructor
+
+    constructor() {
+        return new Proxy(this, {
+            // overrides getting this[index]
+            get(self, index) {
+                if (typeof index === 'symbol') return self[index as any];
+                const i = Number.parseInt(index);
+                if (isNaN(i)) return self[index as any];
+                else return self.at(i);
+                // TODO See how this proxy thing affects performance (CIF parsing itself uses custom parseInt because Number.parseInt is slow)
```

Comment:
The CIF parser relies on `charCodeAt`. We could have an `isASCII` flag, which would return this from the underying `Uint8Array` directly too when available.

Reply:
In the meantime i removed support for [i], in favor of `charAt` and `charCodeAt`.